Xmartlabs' first hackathon post (#89)
xmartlabs/blog: _posts/2019-08-20-first-xmartlabs-hackathon.md

---
layout: post

> title: "Xmartlabs' first hackathon!"

Reviewer: I would personally like more a more impactful title like: 'Diving into the internal Hackathon's world'  or 'First Internal Hackaton Day' or something better. 

Probably someone with a better English than mine could help more for sure.

Reviewer 2: I'm not the best for titles but maybe this sparks someone's imagination

> Venturing into our first Hackathon!

> Delving into our first Hackathon!

> Our very first Hackathon!

> Our very first Hackathon: An enlightening experience!

Author: I like the third one best, but I'm more of a classic guy, so naturally I'm not the best when it comes to titles either :$

---
layout: post
title: "Xmartlabs' first hackathon!"
date: 2019-08-18 17:00:00
author: Bruno Ferrari
categories: hackathon, team, xmartlabs
author_id: bruno
featured_position: 1
featured_image: /images/xmartlabs-first-hackathon/general_view_bbq.jpg
---

Last Saturday Xmartlabs held its very first hackathon!
For those who are not familiar with the concept, a hackathon is an event where everyone who is involved in software development -be it graphic designers, project managers, software developers, etc- come together to collaborate in projects of different nature.

> This means that projects don't necessarily have to be software-related, but can be anything the team sets their mind to.

Author: It's an idiom, https://idioms.thefreedictionary.com/set+their+minds+to

> This means that projects don't necessarily have to be software-related, but can be anything the team sets their mind to.

Reviewer 2: My bad. I'd still suggest using `they` instead of `but`

Our hackathon had teams comprised of up to 6 members, and a whole day duration.

## Projects

In this case, all of the hackaton's projects were software-related (it runs through our veins, I guess).

### IoT

> 

Reviewer 2: ```suggestion

Many have been the days when someone had to go up the stairs resigned of every bathroom being occupied. 

```

> The first project was an IoT project. The team wanted to be able to ask, through a Slack command, if a specific bathroom in our office was occupied or not.

Reviewer 2: ```suggestion
In the field of IoT (Internet of Things), this project came from the team wanting to ask a Slack Bot if a specific bathroom in the office is occupied or not.
```

<img width="100%" src="/images/xmartlabs-first-hackathon/iot_project.jpg" /> 
<br />

> For this the team paired a ultrasonic sensor with an ESP8266 chip and also developed a custom command for Slack to give orders to the chip. 

Author: But the ESP8266 has wifi, doesn't it?

> For this the team paired a ultrasonic sensor with an ESP8266 chip (essentially an Arduino Uno with WiFi capabilities) and also developed a custom command for Slack to give orders to the chip.

Reviewer 2: You use this a lot :stuck_out_tongue_closed_eyes: 

```suggestion
Delving in the area of home automatization, the team paired an ultrasonic sensor along with a ESP8266 chip (essentially an Arduino Uno with WiFi capabilities). All this was connected to Slack, allowing the chip to receive orders from Slack.
```

> Although no one in this specific team had extensive experience working with IoT components, the project was functional in the end.

Reviewer 2: ```suggestion
Even though no one on the team had extensive experience working with IoT components, the project was a great way for us to delve into this area and ended up producing a functional product.
```

### Skillset blog

> The second project was to build a Skillset blog of our team members, for internal use.

Reviewer 2: ```suggestion
Motivated by the surge we had upon our ranks, this project consisted of building an internal Skillset blog of our team members. Allowing an interactive way of accessing information about the Xmartlabs team.
```

> <img width="100%" src="/images/xmartlabs-first-hackathon/skillset_blog.jpg" /> 

Reviewer 3: I would crop this image, we don't need all that ceiling

<br />

> The challenge here were two: get specific information about all of Xmartlabs' collaborators skillset (we are around 50 now, so it isn't an easy task) and build the blog using state-of-the-art technologies like GraphQL and Nest.

Reviewer 2: ```suggestion
Two challenges arose from this idea: the former being the gathering and processing of information about all of Xmartlabs' collaborator's skillset (we are around 50 now, so it isn't an easy task), the latter was related to building the blog by delving into state-of-the-art technologies like GraphQL and Nest.
```

### Endless Runner game

> Another project was about creating a endless runner game using Unity and our very own assets. 

Author: I don't know, it's much more complex than Chrome's, so I wouldn't like to compare them

> Another project was about creating a endless runner game using Unity and our very own assets.

Reviewer 2: ```suggestion
More related to leisure and delving into areas not so related to our daily work. This project consisted of creating an endless runner game using Unity and our very own assets.
```

<img width="100%" src="/images/xmartlabs-first-hackathon/endless_runner.jpg" /> 
<br />

> Although currently it's only working for desktop PC's, the team is already planning on making a port for Android or iOS.

Reviewer 2: ```suggestion
Even though the game only works on PC, the team is eager on making a port for Android or iOS.
```

Author: I like it, but maybe `eager about` instead of `eager on`?

> ### Internal Rails app

Reviewer 2: I'd consider move this section above the Endless Runner. So that the two of internal use are close together.
- IoT
- Skillshare
- Rails app
- Runner

Or 
- Skillshare
- Rails app
- IoT
- Runner

are better presented IMO

> The last project was another internal one.

Reviewer 2: ```suggestion
Motivated by adding a useful internal tool that would benefit all Xmartlabs, the last project consisted of developing an internal tool so that time offs can be requested and administered by us.
```

> The team built a Rails app -CMS, backend and frontend- with goodies like Google sign-in only for Xmartlabs' email accounts and a very nice UX/UI overall.

Reviewer 2: ```suggestion
The team built a Rails app -CMS, backend and frontend- with goodies like Google sign-in only for Xmartlabs' accounts and a very nice UX/UI overall.
```

<img width="100%" src="/images/xmartlabs-first-hackathon/rails_app.jpg" /> 
<br />

> This app is actually going to save Xmartlabs around $750 a year, as we now won't have to use a paid option. 

Reviewer 4: maybe tell that the pms made that, so it was a good project to back to the right side :smile:

Author: hahahah

> This app is actually going to save Xmartlabs around \$750 a year, as we now won't have to use a paid option.

Reviewer 2: maybe use "could?" I think it would be lying to say that it can be put on production :stuck_out_tongue: 

At least I'd say something that it set up the foundations to said app.

Author: It's on production right now :)

## Barbecue

> To celebrate a very fun and productive day, we had a traditional uruguayan BBQ.

Reviewer 2: In English, nationalities are capitalized. 

```suggestion
To celebrate a very fun and productive day, we had a traditional Uruguayan BBQ.
```

<img width="100%" src="/images/xmartlabs-first-hackathon/general_view_bbq.jpg" /> 
<br />

> <img width="100%" src="/images/xmartlabs-first-hackathon/parrilla_overview.jpg" /> 

Reviewer 3: Personally I would remove this one, and just leave the one with Enrique on it

<br />

<img width="100%" src="/images/xmartlabs-first-hackathon/henry_bbq.jpg" /> 
<br />

> We also welcomed Florencia and Valentina to our team:

Reviewer 2: ```suggestion
We also welcomed Florencia and Valentina into our team:
```

> <img width="100%" src="/images/xmartlabs-first-hackathon/welcome_flo.jpg" /> 

Reviewer 3: I would also crop the side of these ones and put them side by side on desktop

<br />

<img width="100%" src="/images/xmartlabs-first-hackathon/welcome_vale.jpg" /> 
<br />

> All in all, it was a very positive experience and one that we for sure are going to repeat soon!

Reviewer 2: ```suggestion
All in all, the experience a very positive one and with no doubt one that we are eager to repeat soon!
```
